Homework2/ArrayExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Linq;
+
+namespace Homework2
+{
+    public static class ArrayExtension
+    {
+        public static T[] AddAt<T>(this T[] array, T item, int idx)
+        {
+            int newSize = array.Length + 1;
+
+            T[] output = new T[newSize];
+
+            for (int i = 0; i < newSize; i++)
+            {
+                if (i == idx)
+                {
+                    output[i] = item;
+
+                    for (int j = idx; j < array.Length; j++)
+                    {
+                        i++;
+                        output[i] = array[j];
+                    }
+
+                    break;
+                }
+
+                output[i] = array[i];
+            }
+
+            return output;
+        }
+
+        public static T[] AddAtBeginning<T>(this T[] array, T item)
+        {
+            return array.AddAt(item, 0);
+        }
+
+        public static T[] AddAtEnd<T>(this T[] array, T item)
+        {
+            return array.AddAt(item, array.Length);
+        }
+
+        public static T[] RemoveAt<T>(this T[] array, int idx)
+        {
+            int newSize = array.Length - 1;
+
+            T[] output = new T[newSize];
+
+            for (int i = 0; i < newSize; i++)
+            {
+                if (i == idx)
+                {
+                    for (int j = idx + 1; j < newSize; j++)
+                    {
+                        output[i] = array[j];
+                        i++;
+                    }
+
+                    break;
+                }
+
+                output[i] = array[i];
+            }
+
+            return output;
+        }
+
+        public static T[] RemoveAtBeginning<T>(this T[] array)
+        {
+            return array.RemoveAt(0);
```

Comment:
Good work using your general purpose method to implement the specialized methods.

Reply:
Tyvm, I try writing code with SOLID and DRY in mind.